src/main/java/school/lemon/changerequest/java/io/hw2/FileCopyUtilImpl_BytePerByte.java

```diff
@@ -0,0 +1,27 @@
+package school.lemon.changerequest.java.io.hw2;
+
+import java.io.*;
+import java.nio.file.FileAlreadyExistsException;
+
+/**
+ * Created by lera on 13.02.17.
+ */
+public class FileCopyUtilImpl_BytePerByte implements FileCopyUtil {
+    @Override
+    public boolean copy(String src, String dst) throws FileNotFoundException, FileAlreadyExistsException, FileCopyException {
+        File fileSource = new File(src);
+        if (!fileSource.exists()) throw new FileNotFoundException(src);
+        File fileDst = new File(dst);
+        if (fileDst.exists()) throw new FileAlreadyExistsException(dst);
```

Comment:
As we are learning Java language that is absolutely OOP lang - please try to avoid code duplications using some abstractions of higher level.
In you case, there is 4 lines, that are copied from one to another implementation:
```java
File fileSource = new File(src);
if (!fileSource.exists()) throw new FileNotFoundException(src);
File fileDst = new File(dst);
if (fileDst.exists()) throw new FileAlreadyExistsException(dst);
```

You have to create `abstract` class, that will handle this functionality. One appropriate way will be to create abstract `copyInternal` method in abstract class, that will be called from non-abstract `copy` method.
Smth like this:
```java
public abstract class AbstractFileCopyUtil implements FileCopyUtil{
  @Override
  public boolean copy(String src, String dst) throws FileNotFoundException, FileAlreadyExistsException,   FileCopyException{
    //put file existence checks here
    copyInternal(srcFile,dstFile);
  }

  protected abstract copyInternal(File srcFile, File dstFile);
}
```